api/src/main/java/org/openmrs/module/kenyaemr/calculation/library/hiv/EligibleForFastrackPatientsCalculation.java

```diff
@@ -44,55 +51,72 @@
  * longer follow-up intervals (depends on clinician ==>longer than 30 days)
  *
  **/
-public class EligibleForFastrackPatientsCalculation extends AbstractPatientCalculation  {
+public class EligibleForFastrackPatientsCalculation extends AbstractPatientCalculation {
 
     protected static final Log log = LogFactory.getLog(EligibleForFastrackPatientsCalculation.class);
     static ConceptService conceptService = Context.getConceptService();
+    static PatientService patientService = Context.getPatientService();
+
 
     @Override
     public CalculationResultMap evaluate(Collection<Integer> cohort, Map<String, Object> parameterValues, PatientCalculationContext context) {
 
-        Concept TCAdate = Dictionary.getConcept(Dictionary.RETURN_VISIT_DATE);
+        String hivConsultationService = "885b4ad3-fd4c-4a16-8ed3-08813e6b01fa";
+        AppointmentsService appointmentsService = Context.getService(AppointmentsService.class);
         Program hivProgram = MetadataUtils.existing(Program.class, HivMetadata._Program.HIV);
         Set<Integer> alive = Filters.alive(cohort, context);
         Set<Integer> inHivProgram = Filters.inProgram(hivProgram, alive, context);
         Set<Integer> ltfu = CalculationUtils.patientsThatPass(calculate(new LostToFollowUpCalculation(), cohort, context));
         CalculationResultMap ret = new CalculationResultMap();
 
-        for(Integer ptId: cohort){
 
+        for(Integer ptId: cohort){
             Integer tcaPlus30days = 0;
-            Integer tcaConcept = 5096;
-            Date tcaDate = null;
             Date tcaObsDate = null;
+            Date appointmentDate = null;
             boolean patientInHivProgram = false;
-
             boolean eligible = false;
-            Date currentDate =new Date();
+            Date currentDate = new Date();
+            AppointmentSearchRequest appointmentSearchRequest = new AppointmentSearchRequest();
```

Comment:
Line 79 and 80 should ideally be outside of the for loop. Please adjust and update the PR

Reply:
That will help if you are evaluating a large cohort of patients